Lib/test/test_ast.py

```diff
@@ -435,13 +437,48 @@ def test_base_classes(self):
         self.assertTrue(issubclass(ast.comprehension, ast.AST))
         self.assertTrue(issubclass(ast.Gt, ast.AST))
 
-    def test_field_attr_existence(self):
-        for name, item in ast.__dict__.items():
+    def test_import_deprecated(self):
+        ast = import_fresh_module('ast')
+        for name in 'Num', 'Str', 'Bytes', 'NameConstant', 'Ellipsis':
+            with self.assertWarnsRegex(DeprecationWarning,
+                    fr'ast\.{name} is deprecated; use ast\.Constant instead'):
+                getattr(ast, name)
+        with self.assertWarnsRegex(DeprecationWarning,
+                r'ast\.slice is deprecated'):
+            ast.slice
+        with self.assertWarnsRegex(DeprecationWarning,
+                r'ast\.Index is deprecated'):
+            ast.Index
+        with self.assertWarnsRegex(DeprecationWarning,
+                r'ast\.ExtSlice is deprecated; use ast\.Tuple instead'):
+            ast.ExtSlice
+
+    def test_field_attr_existence_deprecated(self):
+        with warnings.catch_warnings():
+            warnings.filterwarnings('ignore', '', DeprecationWarning)
+            from ast import Num, Str, Bytes, NameConstant, Ellipsis
+            from ast import slice, Index, ExtSlice
+
+        for name in ('Num', 'Str', 'Bytes', 'NameConstant', 'Ellipsis',
+                     'slice', 'Index', 'ExtSlice'):
+            item = getattr(ast, name)
             if self._is_ast_node(name, item):
                 if name == 'Index':
                     # Index(value) just returns value now.
                     # The argument is required.
                     continue
+                with self.subTest(item):
+                 with self.assertWarns(DeprecationWarning):
+                    x = item()
```

Comment:
Fix `make patchcheck`:
```suggestion
                with self.subTest(item):
                    with self.assertWarns(DeprecationWarning):
                        x = item()
```